python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -0,0 +1,74 @@
+def validate_setup():
+    from .gpu import get_gpu_device_count
+
+    gpus_count = get_gpu_device_count()
+
+    if gpus_count > 0:
+        # Cupy throws RunTimeException to get GPU count,
+        # hence obtaining GPU count by in-house cpp api above
+        import cupy
+
+        # 75 - Indicates to get "cudaDevAttrComputeCapabilityMajor" attribute
+        # 0 - Get GPU 0
+        major_version = cupy.cuda.runtime.deviceGetAttribute(75, 0)
+
+        if major_version >= 6:
+            # You have a GPU with NVIDIA Pascal™ architecture or better
+            # Hardware Generation	Compute Capability
+            #    Turing	                7.5
+            #    Volta	                7.x
+            #    Pascal	                6.x
+            #    Maxwell	            5.x
+            #    Kepler	                3.x
+            #    Fermi	                2.x
+            pass
+        else:
+            from cudf.errors import UnSupportedGPUError
+
+            raise UnSupportedGPUError(
+                "You will need a GPU with NVIDIA Pascal™ architecture or \
+                    better"
+            )
```

Comment:
I'm not sure if we want to explicitly raise an exception here since it's unclear exactly what works and what doesn't before Pascal architecture. Can we just change this to a warning instead?